compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/AnnotationReference.kt

```diff
@@ -81,6 +81,7 @@ public sealed class AnnotationReference {
 
   public abstract fun replaces(parameterIndex: Int = replacesIndex(fqName)): List<ClassReference>
   public abstract fun exclude(parameterIndex: Int = excludeIndex(fqName)): List<ClassReference>
+  public abstract fun modules(parameterIndex: Int = modulesIndex(fqName)): List<ClassReference>
```

Comment:
I initially added this to replace some "modules" lookups in `ModuleMerger` but ended up reverting to what `ModuleMerger` was already doing because of it needing the lower level types. However I noticed this could replace an existing extension function that was coded for just one case, so I kept this. So you won't see any usages in this PR since it replaces the existing extension usage.